core/src/test/scala/pickling/run/runtime-instantiation-static-picklers.scala

```diff
@@ -0,0 +1,219 @@
+package scala.pickling.runtime
+
+import org.scalatest.FunSuite
+import scala.pickling._, scala.pickling.Defaults._, json._
+
+case class Container[T](value: T)
+
+object RuntimeHelper {
+  /* Instantiate a class for an already generated `Unpickler` */
+  def getInstance[T](className: String): T = {
+    try {
+      val clazz = Class.forName(className)
+      println(s"Create instance of $clazz")
+      clazz.newInstance().asInstanceOf[T]
+    } catch {
+      case ex: Throwable =>
+        scala.concurrent.util.Unsafe.instance.allocateInstance(
+          Class.forName(className)
+        ).asInstanceOf[T]
+    }
+  }
+}
+
+class StaticJavaRuntime extends FunSuite {
+
+  implicit val staticOnly = static.StaticOnly
+
+  test("runtime instantiated pickler should work") {
+
+    val c = Container(List(1,2,3,4,5))
+    val containerPickler = implicitly[Pickler[Container[List[Int]]]]
+    val pickled = c.pickle.value
+
+    val picklerClassName = containerPickler.getClass.getName
+    val runtimePickler = RuntimeHelper.getInstance[Pickler[Container[Any]]](picklerClassName)
```

Comment:
I have a use case in which I need to pickle an entity whose type parameter is unknown (imagine that a client sends a message Msg[T] to a server and that server has to reply with Reply[T]). The only approach to deal with this is to pass the pickler class name of `Reply[T]` into `Msg[T]` since in the server we don't know the type of `T` and there's no way we can get it from the unpickled `Msg[T]`. This is exactly what I'm trying to replay here.